test/e2e/azure_apiserver_ilb.go

```diff
@@ -0,0 +1,463 @@
+	By("Getting azureLoadBalancerClient")
+	azureLoadBalancerClient, err := armnetwork.NewLoadBalancersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Verifying the Azure API Server Internal Load Balancer is created")
+	groupName := os.Getenv(AzureResourceGroup)
+	fmt.Fprintf(GinkgoWriter, "Azure Resource Group: %s\n", groupName)
+	internalLoadbalancerName := fmt.Sprintf("%s-%s", input.ClusterName, "public-lb-internal")
+
+	backoff := wait.Backoff{
+		Duration: 300 * time.Second,
+		Factor:   0.5,
+		Jitter:   0.5,
+		Steps:    7,
+	}
+	retryFn := func(ctx context.Context) (bool, error) {
+		defer GinkgoRecover()
+		resp, err := azureLoadBalancerClient.Get(ctx, groupName, internalLoadbalancerName, nil)
+		if err != nil {
+			return false, err
+		}
+
+		By("Verifying the Azure API Server Internal Load Balancer is the right one created")
+		internalLoadbalancer := resp.LoadBalancer
+		Expect(ptr.Deref(internalLoadbalancer.Name, "")).To(Equal(internalLoadbalancerName))
+
+		By("Verifying the Azure API Server Internal Load Balancer is in a succeeded state")
+		switch ptr.Deref(internalLoadbalancer.Properties.ProvisioningState, "") {
+		case armnetwork.ProvisioningStateSucceeded:
+			return true, nil
+		case armnetwork.ProvisioningStateUpdating:
+			// Wait for operation to complete.
+			return false, nil
+		default:
+			return false, fmt.Errorf("azure internal loadbalancer provisioning failed with state: %q", ptr.Deref(internalLoadbalancer.Properties.ProvisioningState, "(nil)"))
+		}
+	}
+	err = wait.ExponentialBackoffWithContext(ctx, backoff, retryFn)
+	Expect(err).NotTo(HaveOccurred())
+
+	// ------------------------ //
+	By("Creating a dynamic client for accessing custom resources in the management cluster")
+	mgmtRestConfig := input.BootstrapClusterProxy.GetRESTConfig()
+	mgmtDynamicClientSet, err := dynamic.NewForConfig(mgmtRestConfig)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(mgmtDynamicClientSet).NotTo(BeNil())
+
+	By("Getting the AzureCluster using the dynamic client set")
+	azureClusterGVR := schema.GroupVersionResource{
+		Group:    "infrastructure.cluster.x-k8s.io",
+		Version:  "v1beta1",
+		Resource: "azureclusters",
+	}
+
+	azureCluster, err := mgmtDynamicClientSet.Resource(azureClusterGVR).
+		Namespace(input.Namespace.Name).
+		Get(ctx, input.ClusterName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	deployedAzureCluster := &infrav1.AzureCluster{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(
+		azureCluster.UnstructuredContent(),
+		deployedAzureCluster,
+	)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Getting the controlplane endpoint name")
+	controlPlaneEndpointDNSName, apiServerILBPrivateIP := "", ""
+	for _, frontendIP := range deployedAzureCluster.Spec.NetworkSpec.APIServerLB.FrontendIPs {
+		if frontendIP.PublicIP != nil && frontendIP.PublicIP.DNSName != "" {
+			fmt.Fprintf(GinkgoWriter, "Control Plane Endpoint Name: %s\n", frontendIP.PublicIP.DNSName)
```

Comment:
nit: Might be good to use Logf instead of fmt.Fprintf here, as well as in the rest of the file.

Reply:
`fmt.Fprintf(GinkgoWriter, .....)` helped me log the output inline with the test. 
Do you know if `Logf()` achieves the same ?

Reply:
It should achieve the same and I've always used Logf to print stuff in-line. Logf under the hood is fmt.Fprintf as seen here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/61c6f3f180539609090fbf5a943b2e021cfbf129/test/e2e/log.go#L36

It seems like Logf is the preferred way to print stuff in E2E tests as it prints the log level and timestamp

Reply:
Updating

Reply:
Updated, please take a look.